src/app/pages/dashboard/types/widget-group.interface.ts

```diff
@@ -38,6 +40,36 @@ export interface WidgetGroup {
   ];
 }
 
+export interface FormWidgetGroup {
+  layout: WidgetGroupLayout;
+  slots: [
+    (FormWidget | null)?,
+    (FormWidget | null)?,
+    (FormWidget | null)?,
+    (FormWidget | null)?,
+  ];
+}
+
+export function widgetGroupToFormWidgetGroup(widgetGroup: WidgetGroup): FormWidgetGroup {
```

Comment:
Optional change. If you were to determine category from widget type, you would not need to convert and de-convert into another object.